### PR TITLE
Infra CLI refactor

### PR DESCRIPTION
**Pull request by csrwng**
- Cleans up packages of CLI code
- Makes infra creation idempotent
- Splits out IAM creation into its own command:
  ```
  oc create infra aws -- creates per-cluster infra
  oc create iam aws -- creates worker instance profile unassociated to a specific cluster
  ```

Still to come (in a future pr):

- [ ] Add logging
- [ ] Add destroy commands for regular infra and IAM resources

### CODE REVIEW

File: cmd/infra/aws/ec2.go [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+		tags = append(tags, &ec2.Tag{
+			Key:   aws.String("Name"),
+			Value: aws.String(name),
+		})
+	}
+	return tags
+
+}
+
+func waitForCondition(checkFunc func() (bool, error), interval, timeOut time.Duration) error {
```
Comment (ironcladlou) [inline cmd/infra/aws/ec2.go:530]: How is this different from `wait.Poll()`?

Comment (csrwng) [reply cmd/infra/aws/ec2.go:530]: hmm, it's not, I will call that instead, thx

Comment (csrwng) [reply cmd/infra/aws/ec2.go]: fixed

**Pull request merged**